src/System/SshdLint/Check.hs

```diff
@@ -10,52 +10,56 @@ import Data.Char (toLower)
 import qualified Data.Map as Map
 import qualified Data.Set as Set
 
-type RecommendedSettings = Map.Map String [String]
-type ActiveSettings      = Map.Map String [String]
+type Configuration = [String]
+type RecommendedSettings = Map.Map String [Configuration]
+type ActiveSettings      = Map.Map String Configuration
 
 
-defaultAcceptedValues :: [(String, [String])]
+defaultAcceptedValues :: [(String, [Configuration])]
 defaultAcceptedValues =
-  [ ("PermitEmptyPasswords", ["no"])
-  , ("PasswordAuthentication", ["no"])
-  , ("HostbasedAuthentication", ["no"])
-  , ("PermitRootLogin", ["no"])
-  , ("IgnoreRhosts", ["yes"])
-  , ("Protocol", ["2"])
-  , ("StrictModes", ["yes"])
-  , ("UsePrivilegeSeparation", ["yes"]) ]
+  [ ("PermitEmptyPasswords", [["no"]])
+  , ("PasswordAuthentication", [["no"]])
+  , ("HostbasedAuthentication", [["no"]])
+  , ("PermitRootLogin", [["no"], ["without-password"]])
+  , ("IgnoreRhosts", [["yes"]])
+  , ("Protocol", [["2"]])
+  , ("StrictModes", [["yes"]])
+  , ("UsePrivilegeSeparation", [["yes"]]) ]
 
 duplicatedValues :: [String] -> Set.Set String
 duplicatedValues values =
   snd $ foldr checkDuplicate (Set.empty, Set.empty) values
 
-activeSettings :: [(String, [String])] -> ActiveSettings
+activeSettings :: [(String, Configuration)] -> ActiveSettings
 activeSettings =
   foldr registerSetting Map.empty
 
-  where registerSetting (configOption, value) =
-          Map.insert (map toLower configOption) value
+  where registerSetting (configOption, values) =
+          Map.insert (map toLower configOption) values
 
 -- | Given a collection of recommendation strings and a current
 -- setting, returns a new list of recommendations with new
 -- recommendations added depending on the setting of the given
 -- configOption and value.
-checkSafeSetting :: Map.Map String [String]
+checkSafeSetting :: Map.Map String [[String]]
                  -> String
-                 -> [String]
-                 -> [String]
+                 -> [Configuration]
+                 -> Configuration
                  -> [String]
 checkSafeSetting recs configOption values previousRecommendations =
   let recommendation = Map.lookup configOption recs in
 
   case recommendation of
     Nothing -> previousRecommendations
-    Just r -> if Set.fromList r == Set.fromList values then
-                previousRecommendations
-              else
-                previousRecommendations ++ [ configOption ++ " should be "
-                                             ++ show r ++ ", found " ++
-                                             show values ]
+    Just r  -> do
+      let config = (Set.fromList . head) values
```

Comment:
@juanpaucar I guess it would be safer, but not too much of a problem if it makes no sense to have an empty list as an argument for that function. Right now, it follows the philosophy of the `head` function in that there's a contract not to send an empty list and doing so would be a bug.